src/prop/prop_proof_manager.h

```diff
@@ -169,6 +178,38 @@ class PropPfManager : protected EnvObj
   std::vector<Node> getInputClauses();
   /** Retrieve the clauses derived from lemmas */
   std::vector<Node> getLemmaClauses();
+  /** Return lemmas used in the SAT proof. */
+  std::vector<Node> getUnsatCoreClauses(std::ostream* outDimacs = nullptr);
+  /**
+   * Get minimized assumptions. Returns a vector of nodes which is a
+   * subset of the assumptions (d_assumptions) that appear in the unsat
+   * core. This should be called only an unsat core is available (after
+   * an unsatisfiable check-sat).
+   */
+  std::vector<Node> getMinimizedAssumptions();
+  /**
+   * Calculate a subset of cset that is propositionally unsatisfiable.
+   * If sucessful, return true and store this in uc.
+   *
+   * @param cset The set of formulas to compute an unsat core for
+   * @param uc The set of formulas returned as the unsat core
+   * @param outDimacs If provided, we write the DIMACS output of uc to this
```

Comment:
```suggestion
   * @param outDimacs If provided, we write a DIMACS representation of uc to this
```
One thing one wonders about this is if the mapping between terms in `uc` and the variables in outDimacs is relevant elsewhere, so it might be saved in the internal state of this class. Is this the case?